types/controller/forwarding/hyperlane.go

```diff
@@ -0,0 +1,152 @@
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN "AS IS" BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package forwarding
+
+import (
+	"fmt"
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/noble-assets/orbiter/types/core"
+)
+
+const (
+	HypTokenIDLen         = 32
+	HypRecipientLen       = 32
+	HypCustomHookLen      = 32
+	HypHookMetadataPrefix = "0x"
+	HypNobleTestnetDomain = 1196573006
+	HypNobleMainnetDomain = 1313817164
+)
+
+// NewHyperlaneAttributes creates and validates new Hyperlane forwarding attributes.
+// Returns an error if validation fails.
+func NewHyperlaneAttributes(
+	tokenID []byte,
+	destinationDomain uint32,
+	recipient []byte,
+	customHookID []byte,
+	customHookMetadata string,
+	gasLimit math.Int,
+	maxFee sdk.Coin,
+) (*HypAttributes, error) {
+	attr := &HypAttributes{
+		TokenId:            tokenID,
+		DestinationDomain:  destinationDomain,
+		Recipient:          recipient,
+		CustomHookId:       customHookID,
+		CustomHookMetadata: customHookMetadata,
+		GasLimit:           gasLimit,
+		MaxFee:             maxFee,
+	}
+
+	if err := attr.Validate(); err != nil {
+		return nil, err
+	}
+
+	return attr, nil
+}
+
+// Validate performs validation on the Hyperlane attributes.
+func (a *HypAttributes) Validate() error {
+	if a == nil {
+		return core.ErrNilPointer.Wrap("Hyperlane attributes are not set")
+	}
+
+	if len(a.TokenId) != HypTokenIDLen {
+		return fmt.Errorf(
+			"token ID must be %d bytes, received %d bytes",
+			HypTokenIDLen,
+			len(a.TokenId),
+		)
+	}
+
+	if len(a.Recipient) != HypRecipientLen {
+		return fmt.Errorf(
+			"recipient must be %d bytes, received %d bytes",
+			HypRecipientLen,
+			len(a.Recipient),
+		)
+	}
+
+	if l := len(a.CustomHookId); l != 0 && l != HypCustomHookLen {
+		return fmt.Errorf(
+			"custom hook ID must be %d bytes when set, received %d bytes",
+			HypCustomHookLen,
+			l,
+		)
+	}
+
+	if a.DestinationDomain == HypNobleMainnetDomain ||
+		a.DestinationDomain == HypNobleTestnetDomain {
+		return fmt.Errorf("destination domain %d is a Noble domain", a.DestinationDomain)
+	}
+
+	if a.CustomHookMetadata != "" &&
+		!strings.HasPrefix(a.CustomHookMetadata, HypHookMetadataPrefix) {
+		return fmt.Errorf(
+			"hook metadata must have the %s prefix, got: %s",
+			HypHookMetadataPrefix,
+			a.CustomHookMetadata,
+		)
+	}
+
+	return nil
+}
+
+var _ core.ForwardingAttributes = &HypAttributes{}
+
+// CounterpartyID returns a string representation of the destination domain.
+func (a *HypAttributes) CounterpartyID() string {
+	return fmt.Sprintf("%d", a.GetDestinationDomain())
```

Comment:
nit: should be more optimized to use `strconv` here instead of formatting

```suggestion
	return strconv.FormatUint(uint64(a.GetDestinationDomain()), 10)
```